workflow-templates/java-pr-pipeline.yml

```diff
@@ -7,6 +10,11 @@ permissions:
 on:
   workflow_dispatch:
   pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
```

Comment:
I understand the first 3 types are default. But why do we need to trigger this on `closed` events?

Reply:
So that when a PR is merged or closed (both are considered `closed`) the preview environment is deleted

Reply:
no, actually, still confused. Wouldn't this make the pipeline run all the jobs besides the `deploy-pr-environment` when one closes the PR?

Reply:
Aha, in the node flow I've done https://github.com/brandwatch/bw-workflow-actions/blob/dev/.github/workflows/node-pull-request.yaml#LL64 which makes none of the dependencies of package-manager run when the pr is being closed.

For Java, this is missing! Nice catch :)